substrate-node/pallets/pallet-smart-contract/src/types.rs

```diff
@@ -229,3 +230,38 @@ pub struct Provider<AccountId> {
     pub who: AccountId,
     pub take: u8,
 }
+
+pub const MAX_METADATA_LENGTH: u32 = 64; // limited to 64 bytes (2 public keys)
+pub const MAX_BILL_METADATA_LENGTH: u32 = 50; // limited to 50 bytes for now
+
+#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct ServiceContract<T: Config> {
+    pub service_twin_id: u32,
+    pub consumer_twin_id: u32,
+    pub base_fee: u64,
+    pub variable_fee: u64,
+    pub metadata: BoundedVec<u8, ConstU32<MAX_METADATA_LENGTH>>,
+    pub accepted_by_service: bool,
+    pub accepted_by_consumer: bool,
+    pub last_bill: u64,
+    pub state: ServiceContractState,
+    pub phantom: PhantomData<T>,
```

Comment:
You added this because the T is not used, right? If it is not used we can remove the parameter... No?

Reply:
It is not used here
but requested in implementation here: https://github.com/threefoldtech/tfchain/blob/dc71ed18fc9cc0b1bf17bebe176b6220460611d0/substrate-node/pallets/pallet-smart-contract/src/cost.rs#L114

So I did not managed to get rid of it...

Reply:
You can add that parameter to the function that needs it instead?

Reply:
It worked, thanks!